disnake/onboarding.py

```diff
@@ -0,0 +1,239 @@
+        emoji = PartialEmoji._emoji_from_name_id(
+            name=name,
+            id=int(id) if id else None,
+            animated=animated,
+            state=guild._state,
+        )
+
+        self = cls(
+            title=data["title"],
+            description=data["description"],
+            emoji=emoji,
+            roles=[Object(id=role_id) for role_id in data["role_ids"]],
+            channels=[Object(id=channel_id) for channel_id in data["channel_ids"]],
+            guild=guild,
+        )
+        self.id = int(data["id"])
+        return self
+
+    @property
+    def roles(self) -> List[Role]:
+        """List[:class:`Role`]: A list of roles that will be added to the user when they select this option."""
+        return list(filter(None, map(self.guild.get_role, self._roles_ids)))
+
+    @property
+    def channels(self) -> List[GuildChannel]:
+        """List[:class:`abc.GuildChannel`]: A list of channels that the user will see when they select this option."""
+        return list(filter(None, map(self.guild.get_channel, self._channels_ids)))
+
+
+class OnboardingPrompt(Hashable):
+    """Represents an onboarding prompt.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The onboarding prompt's ID.
+    type: :class:`OnboardingPromptType`
+        The onboarding prompt's type.
+    options: List[:class:`OnboardingPromptOption`]
+        The onboarding prompt's options.
+    title: :class:`str`
+        The onboarding prompt's title.
+    single_select: :class:`bool`
+        Whether users are limited to selecting one option for the prompt.
+    required: :class:`bool`
+        Whether the prompt is required before a user completes the onboarding flow.
+    in_onboarding: :class:`bool`
+        Whether the prompt is present in the onboarding flow.
+        If `false`, the prompt will only appear in the Channels & Roles tab
+    """
+
+    __slots__ = ("id", "title", "options", "single_select", "required", "in_onboarding", "type")
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        options: List[OnboardingPromptOption],
+        single_select: bool,
+        required: bool,
+        in_onboarding: bool,
+        type: OnboardingPromptType,
+    ):
+        self.id = 0
+        self.title = title
+        self.options = options
+        self.single_select = single_select
+        self.required = required
+        self.in_onboarding = in_onboarding
+        self.type = type
+
+    def __str__(self) -> str:
+        return self.title
+
+    def __repr__(self) -> str:
+        return (
+            f"<OnboardingPrompt id={self.id!r} title={self.title!r} options={self.options!r}"
+            f" single_select={self.single_select!r} required={self.required!r}"
+            f" in_onboarding={self.in_onboarding!r} type={self.type!r}>"
+        )
+
+    @classmethod
+    def _from_dict(cls, *, data: OnboardingPromptPayload, guild: Guild) -> Self:
```

Comment:
`_from_dict` here and in `OnboardingPromptOption` can be merged into `__init__`, now that these don't have to be user-constructible

Reply:
you're right, my bad.

Reply:
7466f3a7aeff7868b8ef8c84e078ca0fd523cb60